core/domain/classifier_services.py

```diff
@@ -444,6 +444,8 @@ def load_examples(self, examples):
 
     def predict_label_for_doc(self, d):
         """Returns the predicted label from a doc's prediction report."""
+        if self._num_docs < 20 or self._num_labels < 3:
```

Comment:
Where do these numbers come from? Might be better to throw them into private globals inside of classifier_services.

Also, this seems a bit fishy. I think I understand what you're trying to do here (and why it might work), but I think it definitely warrants a comment explaining it.